enhancements/network/user-defined-network-segmentation.md

```diff
@@ -413,6 +688,235 @@ guarantee uniqueness and eliminates the ability to falsify access.
 After creating the CRD, check the status to ensure OVN-Kubernetes has accepted this network to serve the namespaces
 selected. Now tenants may go ahead and be provisioned their namespace.
 
+#### Controller
+
+New controller should be introduced, it should manage the new CRDs lifecycle.
+
+The controller should utilize the CNI plugin API and create OVN networks using `NetworkAttachmentDefinition` using the 
+`ovn-k8s-overlay-cni` CNI plugin according to desired spec.
+
+The controller should watch NAD object in order to reconcile them and reflect the network state in the CRDs status.
+
+The request corresponding NADs should be created with a finalizer, so that upon deletion OVN-Kubernetes can validate that 
+there are no pods still using this network.
+If there are pods still attached to this network, the network will not be removed.
```

Comment:
What reporting can you introduce to make this obvious to an end user?

Reply:
there will be a UDN status saying which pods prevent UDN from being deleted with namespace/name